src/esp/sensor/CameraSensor.cpp

```diff
@@ -28,18 +39,17 @@ CameraSensor::CameraSensor(scene::SceneNode& cameraNode,
   renderCamera_->setViewport(this->framebufferSize());
 }  // ctor
 
-void CameraSensor::setProjectionParameters(const SensorSpec::ptr& spec) {
+void CameraSensor::setProjectionParameters(const CameraSensorSpec::ptr& spec) {
   ASSERT(spec != nullptr);
   // update this sensor's sensor spec to reflect the passed new values
   spec_->resolution = spec->resolution;
-  for (const auto& elem : spec->parameters) {
-    spec_->parameters.at(elem.first) = elem.second;
-  }
+  spec_->near = spec->near;
+  spec_->far = spec->far;
 
   width_ = spec_->resolution[1];
   height_ = spec_->resolution[0];
-  near_ = std::atof(spec_->parameters.at("near").c_str());
-  far_ = std::atof(spec_->parameters.at("far").c_str());
+  near_ = spec_->near;
```

Comment:
all these member variables such as near_, height_, width_ in the CameraSensor can be deprecated. 

Before if you do not have them, you need to convert them from string every time you query these values. It is not the case any more.